fortuna/prob_model/posterior/sgmcmc/cyclical_sgld/cyclical_sgld_approximator.py

```diff
@@ -0,0 +1,49 @@
+from typing import Optional
+
+from fortuna.prob_model.posterior.base import PosteriorApproximator
+from fortuna.prob_model.posterior.sgmcmc.sgmcmc_preconditioner import (
+    Preconditioner,
+    identity_preconditioner,
+)
+from fortuna.prob_model.posterior.sgmcmc.cyclical_sgld import (
+    CYCLICAL_SGLD_NAME,
+)
+
+
+class CyclicalSGLDPosteriorApproximator(PosteriorApproximator):
+    def __init__(
+        self,
+        n_samples: int = 10,
+        n_thinning: Optional[int] = 100,
```

Comment:
Also, is 100 a reasonable default? Or would it be too high for expensive models? This is just a question; feel free to convince me that 100 is good.

Reply:
My idea was to support both `None` and 1 as a "disable" value: https://github.com/master/fortuna/blob/sgmcmc/fortuna/prob_model/posterior/sgmcmc/cyclical_sgld/cyclical_sgld_approximator.py#L42

Regarding 100, I don't think there's a one-size-fits-all default, because it really depends on the training budget. OTOH, we already have the default deep ensemble size, which are also model-dependent, so it'd be good to have some sensible value here as well. Perhaps just setting `n_thinning = 1` for now?

Reply:
Makes sense. If in your experiments you find better defaults, we can change it back!